// moongose/firebase_config.js
var admin = require("firebase-admin");
var privateKeyId = process.env.PRIVATE_KEY_ID
var privateKey = process.env.PRIVATE_KEY
var serviceAccount = {
  "type": "service_account",
  "project_id": "dynamic-tide-315303",
  "private_key_id": privateKeyId,
  "private_key": privateKey,
  "client_email": "firebase-adminsdk-cfu7y@dynamic-tide-315303.iam.gserviceaccount.com",
  "client_id": "115147210186055903132",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/firebase-adminsdk-cfu7y%40dynamic-tide-315303.iam.gserviceaccount.com"
}
admin.initializeApp({
  credential: admin.credential.cert(serviceAccount)
});
const sendFirebaseNotification = async (data, title, body) => {
  var payload = {
        notification: {
            title: title,
            body: body,
        }
    };
    var options = {
        priority: "high",
        timeToLive: 60 * 60 *24
    };
    admin.messaging().sendToDevice(data, payload, options)
    .then(function(response) {
        console.log(response);
    })
    .catch(function(error) {
        console.log(error);
    });
}
module.exports = {
    sendFirebaseNotification
}
